src/doc/man/cargo-metadata.md

```diff
@@ -34,7 +34,10 @@ The output has the following format:
             "name": "my-package",
             /* The version of the package. */
             "version": "0.1.0",
-            /* The Package ID, a unique identifier for referring to the package. */
+            /* The Package ID, a unique identifier for referring to the package.
+               Consumers should generally see it as an opaque string. There is
+               no guarantee of the format across different versions of Cargo.
+            */
             "id": "my-package 0.1.0 (path+file:///path/to/my-package)",
```

Comment:
`cargo_metadata` does [make mention](https://docs.rs/cargo_metadata/latest/cargo_metadata/struct.PackageId.html) that `PackageId` is opaque

> An “opaque” identifier for a package. It is possible to inspect the repr field, if the need arises, but its precise format is an implementation detail and is subject to change.

Reply:
A part of me would love for us to exercise this opacity once: to switch from `PackageId` to `PackageIdSpec` which would resolve #7267.